lib/SpeculationModules/SmtxAA.cpp

```diff
@@ -61,7 +61,9 @@ namespace liberty::SpecPriv
                                     TemporalRelation rel, const Value *ptrB,
                                     unsigned sizeB, const Loop *L, Remedies &R,
                                     DesiredAliasResult dAliasRes) {
-    return LoopAA::alias(ptrA, sizeA, rel, ptrB, sizeB, L, R, dAliasRes);
+    //return LoopAA::alias(ptrA, sizeA, rel, ptrB, sizeB, L, R, dAliasRes);
```

Comment:
Similar to SlampOracleAA, we need to convert this to also inherit from the remediator base class. Remove this hack and add a `memdep` that implements the logic without LoopAA.